torch/fx/passes/runtime_assert.py

```diff
@@ -34,6 +44,161 @@ def _get_example_value(node: fx.Node) -> Optional[str]:
+
+    # if straightforward we return
+    if isinstance(expr.args[1], sympy.Number):
+        return expr
+
+    # try flipping, won't hurt
+    # can return if original expr is something like 4 <= s0
+    lhs, rhs = expr.args
+    if isinstance(expr, sympy.LessThan):
+        expr = sympy.GreaterThan(rhs, lhs)
+    elif isinstance(expr, sympy.StrictLessThan):
+        expr = sympy.StrictGreaterThan(rhs, lhs)
+    elif isinstance(expr, sympy.GreaterThan):
+        expr = sympy.LessThan(rhs, lhs)
+    elif isinstance(expr, sympy.StrictGreaterThan):
+        expr = sympy.StrictLessThan(rhs, lhs)
+    elif isinstance(expr, sympy.Equality):
+        expr = sympy.Equality(rhs, lhs)
+    elif isinstance(expr, sympy.Unequality):
+        expr = sympy.Unequality(rhs, lhs)
+    if isinstance(expr.args[1], sympy.Number):
+        return expr
+
+    # move all non-numeric values to LHS
+    eq = expr.args[0] - expr.args[1]
+    non_numeric = tuple(x for x in eq.args if not isinstance(x, sympy.Number))
+    numeric = tuple(x for x in eq.args if isinstance(x, sympy.Number))
+    assert len(numeric) <= 1
+    expr = type(expr)(sympy.Add(*non_numeric), -numeric[0] if numeric else 0)
+
+    # maybe negate
+    lhs, rhs = expr.args
+    if -lhs in mapping:
+        if isinstance(expr, sympy.LessThan):
+            expr = sympy.GreaterThan(-lhs, -rhs)
+        elif isinstance(expr, sympy.StrictLessThan):
+            expr = sympy.StrictGreaterThan(-lhs, -rhs)
+        elif isinstance(expr, sympy.GreaterThan):
+            expr = sympy.LessThan(-lhs, -rhs)
+        elif isinstance(expr, sympy.StrictGreaterThan):
+            expr = sympy.StrictLessThan(-lhs, -rhs)
+        elif isinstance(expr, sympy.Equality):
+            expr = sympy.Equality(-lhs, -rhs)
+        elif isinstance(expr, sympy.Unequality):
+            expr = sympy.Unequality(-lhs, -rhs)
+
+    return expr
+
+
+def _get_last_placeholder(graph: fx.Graph) -> fx.Node:
+    """
+    find location of last placeholder node, as insert point.
+    if not found return first node
+    """
+    res = None
+    for node in graph.nodes:
+        if node.op == "placeholder":
+            res = node
+        else:
+            break
+    if res is None:
+        return next(iter(graph.nodes))
+    return res
+
+
+def _is_size_function_node(node: fx.Node) -> bool:
+    return (
+        node.op == "call_function"
+        and node.target in [torch.ops.aten.sym_size.int, "size"]
+    )
+
+
+def _dce_from_node(node: fx.Node):
+    """
+    DCE unused node and any unused ancestors
+    """
+    graph = node.graph
+    _args = [node]
+    while _args:
+        node = _args.pop()
+        inputs = node.args
+        log.debug("DCE node %s", node)
+        graph.erase_node(node)
+        _args.extend(
+            n for n in inputs
+            if (
+                isinstance(n, fx.Node)
+                and not n.users
+                and n.op != "placeholder"
+            )
+        )
```

Comment:
This looks unsafe. Because runtime assertion insertion can happen in graphs that haven't been functionalized, you cannot in general DCE arbitrary Tensor compute, because that computation may have side effects. You can only DCE "size-like" compute on SymInt/int, which is guaranteed to be purely functional as int is immutable.